be/src/storage/storage_engine.cpp

```diff
@@ -448,14 +448,21 @@ std::vector<DataDir*> StorageEngine::get_stores_for_create_tablet(TStorageMedium
         for (auto& it : _store_map) {
             if (it.second->is_used()) {
                 if (_available_storage_medium_type_count == 1 || it.second->storage_medium() == storage_medium) {
-                    stores.push_back(it.second);
+                    if (it.second->available_bytes() > config::storage_flood_stage_left_capacity_bytes) {
+                        stores.push_back(it.second);
+                    }
                 }
             }
         }
     }
+
+    std::sort(stores.begin(), stores.end(),
+              [](const auto& a, const auto& b) { return a->available_bytes() > b->available_bytes(); });
+
+    const int mid = stores.size() / 2 + 1;
```

Comment:
Why use `stores.begin() + mid` instead of `stores.end()`?

Reply:
> Why use `stores.begin() + mid` instead of `stores.end()`?

@chaoyli Thank you for your comments. After sorting the disk by available space (line 459 to 460), I'm intend to randomly choose one of the first half of disks, instead of shuffle the entire disk.